### DISCUSSION TITLE
How to finetune the LoFTR model with my own dataset?

### DISCUSSION THREAD

**Discussion by GivanTsai** [kornia/kornia#1699] [Questions] [answered]
How to finetune the LoFTR model with my own dataset?

**Comment by ducha-aiki** [comment, marked as answer]
Unfortunately the best way is to use training scripts from original LoFTR repo https://github.com/zju3dv/LoFTR

**Reply by GivanTsai** [reply]
> iginal LoFTR repo https://github.com/zju3dv/LoFTR

If I finetune the model using this repo and get the new weight file, can I also use the kornia to do the inference?

**Reply by edgarriba** [reply]
That would a good thing to check. I remember we made minor tweaks to pass our testing infra to run that model in "production" safe environments. That's usually the problem of mostly all the research code projects, that it needs a lot of curation to be exported later to onnx, jit or tensorrt and usually the tools used to train models are totally far away from that phase (even pytorch itself still has major issues to make things things properly in production).

Said that, one quick way to check that compatibility is still working with kornia and the original code would be just to take the original repo in the last commit, make a run, save weights and from Kornia API manually load those weights.

**Reply by adityamwagh** [reply]
What I have in mind is that I will use the model architecture in kornia repo, and write a separate training script for it. Any thing that I should keep in mind?

**Reply by edgarriba** [reply]
@adityamwagh would be great if you could use our training api. Can be a way also to get feedback and improve. In the end it's just a light abstraction of bare metal pytorch training/validation structure.
